[CZBANK-73] feat: added array of BAs with choice on which BA to generate transactions

silviczka opened the pull request:
- create array of BAs for seed-users
- select BA with index 0 as primary BA to have seeded balance value
-set constant PRIMARY_TRANSACTION_INDEX to determine which BA in array will have seeded generated transactions, atm set to 1
-simplify seed-users by extracting functions to seed-helpers folder

*note: seems like CUID in postgreSQL isn't chronologic, so it can't be used to select 1st BA number, that's why I used strategy to use index of bankAccountNumber in array of seeded users

silviczka committed:
feat: added array of BAs with choice on which BA to generate transactions

vojtech-cerveny commented on scripts/seed-helpers/ba-utils.ts:
```diff
@@ -0,0 +1,108 @@
+import bankAccountService from "@/domain/bankAccount-domain/ba-service";
+import { Currency, PrismaClient } from "@prisma/client";
+
+export default async function ensureUserBankAccounts(prisma: PrismaClient, user: any, userSeed: any) {
```
**question:** can you add types here instead of `any` please?

vojtech-cerveny replied:
**todo:** Please, can you also add some "documentation" what it is doing? Like JSDoc, you know.

vojtech-cerveny reviewed (approved):
If you can add some typing, it would be great. Otherwise it looks good, a lot of work is done here, thanks for that! It's great!

silviczka committed:
chore: add JSdocs, fix type any, fix unused vars

silviczka commented:
Hello, 
just pushed new commit, pls let me know if you'd like me to fix anything, or you can fix/rewrite/merge as you need.😀
Hope I didn't break any logic, if I did, sorry! 😅

-add JSdocs

-fix type any

-removed unused vars primaryTransactionIndex and primaryBalanceIndex, they were initially implemented to allow set each users primary balance BA and primary transaction BA individually, but in the end introduced strategy to make global constants (PRIMARY_TRANSACTION_INDEX, PRIMARY_BALANCE_INDEX) which would set globally index of BA in array for transactions and for balance value, if index is higher than number of BAs, fallback is set up to avoid fail and error

-exclude manually created users from seeding transactions - I have noticed in logs (during seeding users) manually created users from my local docker DB appearing in generated transactions, but it didn't really write transactions into these user accounts, so the logs were misleading, fixed now

-some minor logs were added to notify us during seeding to prevent something fail silently - can be removed if useless
